src/Microsoft.AspNet.Razor/ParserResults.cs

```diff
@@ -31,40 +40,50 @@ public class ParserResults
         /// Instantiates a new <see cref="ParserResults"/> instance.
         /// </summary>
         /// <param name="success"><c>true</c> if parsing was successful, <c>false</c> otherwise.</param>
-        /// <param name="document">The Razor syntax tree.</param>
-        /// <param name="tagHelperDescriptors"><see cref="TagHelperDescriptor"/>s that apply to the current Razor 
-        /// document.</param>
-        /// <param name="errors"><see cref="RazorError"/>s encountered when parsing the current Razor
-        /// document.</param>
+        /// <param name="document">The <see cref="Block"/> for the syntax tree.</param>
+        /// <param name="tagHelperDescriptors">
+        /// The <see cref="TagHelperDescriptor"/>s that apply to the current Razor document.
+        /// </param>
+        /// <param name="errorSink">
+        /// The <see cref="ParserErrorSink"/> used to collect <see cref="RazorError"/>s encountered when parsing the
+        /// current Razor document.
+        /// </param>
         protected ParserResults(bool success,
                                 [NotNull] Block document,
                                 [NotNull] IEnumerable<TagHelperDescriptor> tagHelperDescriptors,
-                                [NotNull] IList<RazorError> errors)
+                                [NotNull] ParserErrorSink errorSink)
         {
             Success = success;
             Document = document;
             TagHelperDescriptors = tagHelperDescriptors;
-            ParserErrors = errors ?? new List<RazorError>();
+            ErrorSink = errorSink;
+            ParserErrors = errorSink.Errors?.ToList() ?? new List<RazorError>();
```

Comment:
@NTaylorMullen I'll simplify this code too.  either `(errorSink.Errors as IList<RazorError>) ?? errorSink.Errors.ToList()` (because the second part will never execute today) or just `errorSink.Errors.ToList()`.

but why is `ParserErrors` an `IList<RazorError>` and not `IEnumerable<RazorError>`?  no apparent reason to make `ParserErrors` writeable.  the collection is only read twice, including the `Any()` call above (replacing the `Count()` call).  the underlying collection is a `List<RazorError>` anyhow.